data/account/participationRegistry.go

```diff
@@ -380,7 +412,12 @@ func (db *participationDB) writeThread() {
 		if len(wr.registerUpdated) != 0 {
 			err = db.registerInner(wr.registerUpdated)
 		} else if !wr.insertID.IsZero() {
-			err = db.insertInner(wr.insert, wr.insertID)
+			if wr.insert != (Participation{}) {
+				err = db.insertInner(wr.insert, wr.insertID)
+			}
+			if len(wr.keys) != 0 {
```

Comment:
You need to add "&& err==nil"

Reply:
Thanks, I didn't realize that this code technically supported inserting a new participation key while adding state proof keys.

The intention is that those would be done separately though, so I'll change this to "else if".